core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -1556,7 +1550,7 @@ object Parser {
     var capture: Boolean = true
   }
 
-  private object Impl {
+  private[parse] object Impl {
```

Comment:
Made this `private[parse]` again so we can use `rangesFor` in tests

Reply:
rather than opening all of Impl, can we lift rangesFor out of Impl and make it `private[parse]`? Maybe irrationally, I prefer to use private as much as possible. I don't recall all the details, but some of the `private[foo]` stuff is actually public in the byte code.

Lastly, I think it can change inlining, but I'm not 100% sure about that.

Reply:
I see your point. The whole access modifier representation in JVM is interesting. 

> private[foo] stuff is actually public in the byte code.

True, and the funny fact is that `private class / object` gets represented as `public`. the access modifier is only represented in attributes under `InnerClasses`. Methods however they do get their `private` modifier (expect the `private[foo]`). This might be even more relevant if Java interop is in mind.

Anyway initially, I was wondering if down the line we will need to tap into `Impl` a bit more for testing but I guess for now moving this one out is fine as well.